ckanext/dcatapchharvest/profiles.py

```diff
@@ -768,16 +789,41 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
             #  Simple values
             items = [
                 ('status', ADMS.status, None, Literal),
-                ('rights', DCT.rights, None, Literal),
                 ('coverage', DCT.coverage, None, Literal),
-                ('license', DCT.license, None, Literal),
                 ('identifier', DCT.identifier, None, Literal),
                 ('media_type', DCAT.mediaType, None, Literal),
                 ('spatial', DCT.spatial, None, Literal),
             ]
 
-            self._add_triples_from_dict(resource_dict, distribution, items)
+            if resource_dict.get('rights'):
+                rights_uri = dh.get_license_uri_by_name(
+                    resource_dict.get('rights')
+                )
+                if rights_uri is not None:
+                    rights_ref = URIRef(rights_uri)
+                    g.add((rights_ref, RDF.type, DCT.RightsStatement))
+                    g.add((distribution, DCT.rights, rights_ref))
+                if rights_uri is None:
+                    resource_rights_ref = URIRef(resource_dict.get('rights'))
```

Comment:
We still have to check that we get a value for `dh.get_license_name_by_uri(resource_dict.get('rights')` here, even though we're not using the name that we get back. We need to check that the rights statement belongs to our `license.ttl` vocabulary before we map it to the distribution.

Reply:
added the check with the following commit. You are right.